app/src/main/java/com/github/se/polyfit/ui/components/button/BookmarkButton.kt

```diff
@@ -29,18 +29,26 @@ fun BookmarkButton(
       checked = isChecked,
       onCheckedChange = {
         isChecked = !isChecked
-        if (isChecked) {
-          addRecipe(recipe)
-        } else {
-          removeRecipe(recipe)
-        }
+        addOrRemoveRecipe(isChecked, recipe, addRecipe, removeRecipe)
       },
       modifier = modifier.testTag("BookmarkButton"),
       colors = IconButtonDefaults.filledIconToggleButtonColors(containerColor = Purple80),
   ) {
     Icon(
-        imageVector = if (isChecked) Icons.Default.Done else Icons.Default.Add,
+        imageVector = Icons.Default.Done.takeIf { isChecked } ?: Icons.Default.Add,
         contentDescription = "Bookmark Button",
         tint = Color.Red)
   }
 }
+
+private fun addOrRemoveRecipe(
+    isChecked: Boolean,
+    recipe: Recipe,
+    addRecipe: (Recipe) -> Unit = {},
+    removeRecipe: (Recipe) -> Unit = {}
+) {
+  when (isChecked) {
```

Comment:
Now that we don't need <50 complexity. Isn't it better to just put an if else?

Reply:
Well i'm thinking it is good practice and if for the next milestone they are stricter on complexity it will be changed anyway